test/format/backup.c

```diff
@@ -53,6 +66,234 @@ check_copy(void)
+        else
+            cmp = strcmp(prev->names[prevpos], cur->names[curpos]);
+
+        if (cmp == 0)
+            curpos++;
+        else if (cmp < 0) {
+            /*
+             * There is something in the prev list not in the current list. Remove it, and continue
+             * - don't advance the current list.
+             */
+            testutil_check(
+              __wt_snprintf(filename, sizeof(filename), "BACKUP/%s", prev->names[prevpos]));
+#if 0
+            fprintf(stderr, "Removing file from backup: %s\n", filename);
+#endif
+            remove(filename);
+            testutil_check(
+              __wt_snprintf(filename, sizeof(filename), "BACKUP_COPY/%s", prev->names[prevpos]));
+            remove(filename);
+        } else {
+            /*
+             * There is something in the current list not in the prev list. Walk past it in the
+             * current list and try again.
+             */
+            curpos++;
+            goto again;
+        }
+    }
+}
+
+/*
+ * active_files_free --
+ *     Free the list of active files.
+ */
+static void
+active_files_free(ACTIVE_FILES *active)
+{
+    uint32_t i;
+
+    if (active == NULL)
+        return;
+    for (i = 0; i < active->count; i++)
+        free(active->names[i]);
+    free(active->names);
+    active_files_init(active);
+}
+
+/*
+ * copy_blocks --
+ *     Perform a single block-based incremental backup of the given file.
+ */
+static void
+copy_blocks(WT_SESSION *session, WT_CURSOR *bkup_c, const char *name)
+{
+    WT_CURSOR *incr_cur;
+    size_t len, tmp_sz;
+    ssize_t rdsize;
+    uint64_t offset, type;
+    u_int size;
+    int ret, rfd, wfd1, wfd2;
+    char buf[512], config[512], *first, *second, *tmp;
+    bool first_pass;
+
+    /*
+     * We need to prepend the home directory name here because we are not using the WiredTiger
+     * internal functions that would prepend it for us.
+     */
+    len = strlen(g.home) + strlen("BACKUP") + strlen(name) + 10;
+    first = dmalloc(len);
+
+    /*
+     * Save another copy of the original file to make debugging recovery errors easier.
+     */
+    len = strlen(g.home) + strlen("BACKUP_COPY") + strlen(name) + 10;
+    second = dmalloc(len);
+    testutil_check(__wt_snprintf(config, sizeof(config), "incremental=(file=%s)", name));
+
+    /* Open the duplicate incremental backup cursor with the file name given. */
+    tmp_sz = 0;
+    tmp = NULL;
+    first_pass = true;
+    rfd = wfd1 = wfd2 = -1;
+    testutil_check(session->open_cursor(session, NULL, bkup_c, config, &incr_cur));
+    while ((ret = incr_cur->next(incr_cur)) == 0) {
+        testutil_check(incr_cur->get_key(incr_cur, &offset, (uint64_t *)&size, &type));
+        if (type == WT_BACKUP_RANGE) {
+            testutil_check(__wt_snprintf(first, len, "%s/BACKUP/%s", g.home, name));
+            testutil_check(__wt_snprintf(second, len, "%s/BACKUP_COPY/%s", g.home, name));
```

Comment:
These two sets of calls actually cannot be moved out of the list. They're different. In the case of the range call, we're using system calls and need to prepend the home directory. In the case of the whole file, we're using WiredTiger calls and WT will prepend the home dir for us. I have added a comment in both places calling out the difference so that it is clear.